server/views/pages/dateAndTime.njk

```diff
@@ -151,6 +151,15 @@
     </form>
 
     {% if slotsList | length %}
+      {% for i in slotsList | list %}
```

Comment:
Can you do this processing in the route and pass in `slotsPresent`? Keeps as much processing out of the templates as possible. Ideally the template just wants to display stuff rather than working out variables. Also use meaningful variable names, `i` could be `slot` as a more descriptive name 👍

Reply:
Could then probably use some shiny array methods like `.find()` etc to do what you want more functionally 🙂